### ISSUE TITLE
cluster_version = "1.21.2" fails but cluster_version = "1.21" works

### ISSUE THREAD

**Issue by timblaktu** [terraform-aws-modules/terraform-aws-eks#1669]
## Description

The latest supported K8s version is 1.21.2, as described [here](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html), however specifying this precise version fails with an "unsupported kubernetes version" error:

```
cluster_version = "1.21.2"
```

whereas specifying a less-specific version appears to work and install 1.21.2:

```
cluster_version = "1.21"
```

## Versions

 - Terraform:
Terraform v1.0.9
on linux_amd64
 - Provider(s):
Terraform v1.0.9
on linux_amd64
```
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.63.0"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "~> 2.6.1"
    }
    helm = {
      source  = "hashicorp/helm"
      version = "~> 2.3.0"
    }
  }
```

 - Module:
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "17.23.0"

## Reproduction

Yes

Yes

### Code Snippet to Reproduce
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "17.23.0"
  cluster_version = "1.21.2"
  <other required inputs>
}

## Expected behavior

Terraform apply succeeds

## Actual behavior

Terraform apply fails to create EKS cluster due to "unsupported kubernetes version", despite the exact version specified being the most recent supported version by AWS/EKS.

**Comment by daroga0002** [comment]
you can pass only a major version, as AWS doesnt support choosing a minor version creation

**Comment by timblaktu** [comment]
Thanks @daroga0002, perhaps the doctoring on the `cluster_version` could be updated to reflect this, that it is only major versions are accepted?

**Comment by daroga0002** [comment]
I have opened a PR to put more clear description over this input.


Adding Kubernetes version policy link for reference
https://kubernetes.io/releases/version-skew-policy/

**Comment by antonbabenko** [comment]
This issue has been resolved in version 17.24.0 :tada:
